net/qBittorrent/Portfile

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:et:sw=4:ts=4:sts=4
+
+PortSystem      1.0
+PortGroup       github 1.0
+PortGroup       qt5 1.0
+
+github.setup    qbittorrent qBittorrent 4.1.9.1 release-
+revision        0
+categories      net
+maintainers     {i0ntempest @i0ntempest} openmaintainer
+categories      net
+platforms       darwin
+license         GPL-2
+
+description     The qBittorrent project aims to provide an open-source software alternative to µTorrent.
+long_description \
+                qBittorrent is a bittorrent client programmed in C++ / Qt that uses \
+                libtorrent (sometimes called libtorrent-rasterbar) by Arvid Norberg. \
+                It aims to be a good alternative to all other bittorrent clients out \
+                there. qBittorrent is fast, stable and provides unicode support as well \
+                as many features.
+
+homepage        https://www.qbittorrent.org
+
+checksums       rmd160  e1a79ed6fb7e014b8bc345f0e7be54274bd1994f \
+                sha256  aa676a985044c88b1cfbeb7f45d4fab27e36de00343431e33ece6791463ee381 \
+                size    7803738
+
+depends_build-append \
+                port:zlib \
+                port:libtorrent-rasterbar11
+
+qt5.depends_component \
+				qttools
+
+configure.args-append \
+				--disable-gui
+
+destroot {
+		if {[variant_isset gui]} {
+                exec ${prefix}/libexec/qt5/bin/macdeployqt ${worksrcpath}/src/qBittorrent.app >/dev/null 2>/dev/null
+                exec /usr/bin/codesign --force --deep --sign - ${worksrcpath}/src/qBittorrent.app
+                copy ${worksrcpath}/src/qBittorrent.app ${destroot}/Applications/MacPorts/
+        } else {
+                exec ${prefix}/libexec/qt5/bin/macdeployqt ${worksrcpath}/src/qbittorrent-nox.app >/dev/null 2>/dev/null
+                exec /usr/bin/codesign --force --deep --sign - ${worksrcpath}/src/qbittorrent-nox.app
+                copy ${worksrcpath}/src/qbittorrent-nox.app ${destroot}/Applications/MacPorts/
+        }
+                
+}
```

Comment:
I haven't seen calling these external programs in destroot before and am not sure if that's an appropriate thing to do.

Reply:
I'm not sure on this either. Since `make install` does nothing here I have to call `macdeployqt` manually. I'll take a look at that PG and see if I can make use of it

Reply:
@reneeotten Sorry I couldn't find a way to destroot the app without calling `macdeployqt`. The app generated by MacPorts does not bundle necessary frameworks and libs so the app won't start. We probably need a way to run `macdeployqt` in qt5 PortGroup. `/usr/bin/codesign` exists on every Mac and is just for silencing firewall dialogs for unsigned apps. If you think that's inappropriate I could make a note telling users to run that command.